internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp

```diff
@@ -135,23 +135,35 @@ ExecPlanNodeVisitor::visit(RetrievePlanNode& node) {
 
     auto active_count = segment->get_active_count(timestamp_);
 
-    if (active_count == 0) {
+    if (active_count == 0 && !node.is_count) {
```

Comment:
When activate_count is 0, add logic for fast return when is_count is True

Reply:
OK, I agree.